extension/httpfs/include/s3fs.hpp

```diff
@@ -19,6 +19,8 @@
 #include <exception>
 #include <iostream>
 
+#undef RemoveDirectory
```

Comment:
curious why do we need the `undef` here?

Reply:
MSVC defines a same macro and would lead to clash.

Reply:
gotcha would appreciate if we could leave a comment for easier understanding :)